neuralpp/symbolic/expression.py

```diff
@@ -476,10 +495,29 @@ def dict(self) -> Dict[str, Any]:
         pass
 
 
+class AbelianOperation(Constant, ABC):
+    """
+    An Abelian operation is a commutative, associative binary operation with an identity element.
+    """
+    @property
+    @abstractmethod
+    def identity(self) -> Expression:
+        """
+        An identity element on a set is an element of the set which leaves unchanged every element of the set
+        when the operation is applied.
+        E.g., for the operation "add" on natural numbers, 0 is the identity element. Because for all x, x + 0 = x.
+        """
+        pass
+
+    @property
+    def element_type(self) -> ExpressionType:
+        return self.identity.type
+
+
 class QuantifierExpression(Expression, ABC):
     @property
     @abstractmethod
-    def operation(self) -> Constant:
+    def operation(self) -> AbelianOperation:
         """ operation is a Constant wrapping a function.
```

Comment:
Very small nit: I think for the other comments, you have """ on a separate line
E.g.:
```
"""
Comment
""""
```